examples/SSVEP/plot_classify_ssvep_mdm.py

```diff
@@ -0,0 +1,239 @@
+destination = _url_to_local_path(url, os.path.join(path, key_dest))
+os.makedirs(os.path.dirname(destination), exist_ok=True)
+if not os.path.exists(destination):
+    _fetch_file(url, destination, print_destination=False)
+
+# Read data in MNE Raw and numpy format
+raw = Raw(destination, preload=True, verbose='ERROR')
+events = find_events(raw, shortest_event=0, verbose=False)
+raw = raw.pick_types(eeg=True)
+event_id = {'13 Hz': 2, '17 Hz': 4, '21 Hz': 3, 'resting-state': 1}
+sfreq = int(raw.info['sfreq'])
+
+eeg_data = raw.get_data()
+
+###############################################################################
+# Visualization of raw EEG data
+# -----------------------------
+#
+# Plot few seconds of signal from the `Oz` electrode using matplotlib
+
+n_seconds = 2
+time = np.linspace(0, n_seconds, n_seconds*sfreq).reshape((1, n_seconds*sfreq))
+plt.figure(figsize=(10, 4))
+plt.plot(time.T, eeg_data[np.array(raw.ch_names) == 'Oz', :n_seconds*sfreq].T,
+         color='C0', lw=0.5)
+plt.xlabel("Time (s)")
+_ = plt.ylabel(r"Oz ($\mu$V)")
+
+###############################################################################
+# And a fewer seconds of all electrodes:
+
+time = np.linspace(0, n_seconds, n_seconds*sfreq).reshape((1, n_seconds*sfreq))
+plt.figure(figsize=(10, 4))
+for ch_idx, ch_name in enumerate(raw.ch_names):
+    plt.plot(time.T, eeg_data[ch_idx, :n_seconds*sfreq].T, lw=0.5, label=ch_name)
+plt.xlabel("Time (s)")
+plt.ylabel(r"EEG ($\mu$V)")
+plt.legend(loc='upper right')
+
+###############################################################################
+# With MNE, it is much easier to visualize the data
+
+raw.plot(duration=n_seconds, start=0, n_channels=8, scalings={'eeg': 4e-2}, 
+         color={'eeg': 'steelblue'})
+
+###############################################################################
+# Extended signals for spatial covariance
+# ---------------------------------------
+# Using the approach proposed by [1]_, the SSVEP signal is extended to include
+# the filtered signals for each stimulation frequency. We define a function to
+# filter the signal:
+
+def _butter_bandpass(signal, lowcut, highcut, fs, order=4):
+    """ Bilateral Butterworth filter for offline filtering """
+    nyq = 0.5 * fs
+    low = lowcut / nyq
+    high = highcut / nyq
+    b, a = butter(order, [low, high], btype='band')
+    filtered = filtfilt(b, a, signal, axis=-1)
+    return filtered
+
+
+# we stack the filtered signals to build an extended signal
+frequencies = [13., 17., 21.]
+freq_band = 0.1
+ext_signal = np.empty_like(eeg_data[0, :])
+for f in frequencies:
+    ext_signal = np.vstack((ext_signal,
+                            _butter_bandpass(eeg_data,
+                                             lowcut=f-freq_band,
+                                             highcut=f+freq_band,
+                                             fs=sfreq)))
+ext_signal = ext_signal[1:, :]
+
+# and we build an array with the signal for each trial
```

Comment:
I think mne can do this by creating a `mne.Epochs` from your events, and you have both of these on mne already.
I always have trouble mne correctly so I can't propose a piece of code.

Reply:
Not sure how to do this in a simple piece of code, but it could be possible to use tailored events. I couldn't figure a piece of code either.

Reply:
i agree, lets try to use the mne.Epochs for this

Reply:
Okay! I forgot that I did not use mne.Epoch at all in this PR. I rewrote the code to use the epochs correctly.